### PR TITLE
python311Packages.scikit-misc: replace ILP64 libs with LP64

### PR DESCRIPTION
**Pull request by ghost**
scikit-misc was using the openblas package which is ILP64, however, numpy uses LP64 as to most of the other python packages. This change lines up scikit-misc with the same BLAS/LAPCAK libs numpy is using.

Also enable the only test.

I fixed this yesterday as part of ZHF https://github.com/NixOS/nixpkgs/pull/268456 but after thinking about it and realizing that pkg `openblas` is ILP64 decided that the better fix is to just align the BLAS/LAPACK libs with numpy.

plotnine fails due to many Attribute errors and the other packages the fail are dependent on plotnine.
Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages marked as broken and skipped:</summary>
  <ul>
    <li>python310Packages.rlax</li>
    <li>python310Packages.rlax.dist</li>
    <li>python311Packages.rlax</li>
    <li>python311Packages.rlax.dist</li>
  </ul>
</details>
<details>
  <summary>8 packages failed to build:</summary>
  <ul>
    <li>python310Packages.bsuite</li>
    <li>python310Packages.bsuite.dist</li>
    <li>python310Packages.plotnine</li>
    <li>python310Packages.plotnine.dist</li>
    <li>python311Packages.bsuite</li>
    <li>python311Packages.bsuite.dist</li>
    <li>python311Packages.plotnine</li>
    <li>python311Packages.plotnine.dist</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.scikit-misc</li>
    <li>python310Packages.scikit-misc.dist</li>
    <li>python311Packages.scikit-misc</li>
    <li>python311Packages.scikit-misc.dist</li>
  </ul>
</details>

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by a-n-n-a-l-e-e** [comment]
@pbsds changed scikit-misc to LP64

File: pkgs/development/python-modules/scikit-misc/default.nix [hunk 1]
```diff
@@ -25,22 +27,14 @@ buildPythonPackage rec {
...
-  ];
-
   postPatch = ''
     patchShebangs .
 
     substituteInPlace pyproject.toml \
       --replace 'numpy==' 'numpy>='
+
+    # remove options that break pytest
+    sed -i '/^addopts/d' pyproject.toml
```
Comment (pbsds) [inline pkgs/development/python-modules/scikit-misc/default.nix:37]:
```suggestion
    # disable coverage testing in pytest
    substituteInPlace pyproject.toml --replace 'addopts = "' '#addopts = "'
```

File: pkgs/development/python-modules/scikit-misc/default.nix [hunk 2]
```diff
@@ -54,9 +48,32 @@ buildPythonPackage rec {
...
+
+  nativeCheckInputs = [
+    pytestCheckHook
+  ];
+
+  # have to move things around to avoid import errors when loading the test
+  preCheck = ''
+    mv skmisc/loess/tests .
+    mv skmisc skm
+  '';
```
Comment (pbsds) [inline pkgs/development/python-modules/scikit-misc/default.nix:73]: Does this work?
```suggestion
  pytestFlagsArray = [ "--import-mode=importlib" ];
```

**Review by pbsds — APPROVED** [review]
Result of `nixpkgs-review pr 268698` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages marked as broken and skipped:</summary>
  <ul>
    <li>python310Packages.rlax</li>
    <li>python310Packages.rlax.dist</li>
    <li>python311Packages.rlax</li>
    <li>python311Packages.rlax.dist</li>
  </ul>
</details>
<details>
  <summary>8 packages failed to build:</summary>
  <ul>
    <li>python310Packages.bsuite</li>
    <li>python310Packages.bsuite.dist</li>
    <li>python310Packages.plotnine</li>
    <li>python310Packages.plotnine.dist</li>
    <li>python311Packages.bsuite</li>
    <li>python311Packages.bsuite.dist</li>
    <li>python311Packages.plotnine</li>
    <li>python311Packages.plotnine.dist</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.scikit-misc</li>
    <li>python310Packages.scikit-misc.dist</li>
    <li>python311Packages.scikit-misc</li>
    <li>python311Packages.scikit-misc.dist</li>
  </ul>
</details>


```
python310Packages.scikit-misc.log:Run-time dependency python found: YES 3.10
python310Packages.scikit-misc.log:Run-time dependency blas found: YES 3
python310Packages.scikit-misc.log:Run-time dependency cblas found: YES 3
python310Packages.scikit-misc.log:Run-time dependency lapack found: YES 3
python311Packages.scikit-misc.log:Run-time dependency python found: YES 3.11
python311Packages.scikit-misc.log:Run-time dependency blas found: YES 3
python311Packages.scikit-misc.log:Run-time dependency cblas found: YES 3
python311Packages.scikit-misc.log:Run-time dependency lapack found: YES 3
```

The broken reverse dependencies are also broken without this PR.

Truly amazing work, thanks for the follow up! Just a couple of nits, then this should be good to merge. 

The 23.11 branchoff is scheduled at 19:00 UTC (2 hours), but if we miss it it is trivial to retarget the pull to the release branch. :)

File: pkgs/development/python-modules/scikit-misc/default.nix (hunk 2 above)
Comment (a-n-n-a-l-e-e) [reply pkgs/development/python-modules/scikit-misc/default.nix:73]: this still results in the error, unfortunately.

```
==================================== ERRORS ====================================
_____________________ ERROR collecting tests/test_loess.py _____________________
skmisc/__init__.py:15: in <module>
    from skmisc.__config__ import show as show_config  # noqa: F401
E   ModuleNotFoundError: No module named 'skmisc.__config__'

During handling of the above exception, another exception occurred:
/nix/store/ghh2dgix6cfhwglxab5aar02y4qa73xb-python3.11-pytest-7.4.3/lib/python3.11/site-packages/_pytest/runner.py:341: in from_call
    result: Optional[TResult] = func()
...
tests/test_loess.py:9: in <module>
    from skmisc.loess import loess, loess_anova
skmisc/__init__.py:20: in <module>
    raise ImportError('\n\n'.join([err.message, msg]))  # type: ignore
E   AttributeError: 'ModuleNotFoundError' object has no attribute 'message'
```

Comment (a-n-n-a-l-e-e) [reply pkgs/development/python-modules/scikit-misc/default.nix:73]: could do something like this.
```
  preCheck = ''
    export HOME="$(mktemp -d)"
    cp -a skmisc/loess/tests "$HOME"
    cd "$HOME"
  '';
```
it is different but not clear it is any better. setuptools has a flag `--inplace` that solved similar problems but that does not seem to work when added to pypaBuildFlags.

Comment (a-n-n-a-l-e-e) [reply pkgs/development/python-modules/scikit-misc/default.nix:72]: this works tho.
```nix
  # can not run tests from source directory
  preCheck = ''
    export HOME="$(mktemp -d)"
    cd
  '';

  pytestFlagsArray = [
    "--pyargs skmisc"
  ];
```

**Comment by a-n-n-a-l-e-e** [comment]
i added version information in `postPatch` as the build was trying to extract it from git. verified `skmisc.__version__` is `0.3.0`

**Pull request merged**